components/scream/src/dynamics/homme/atmosphere_dynamics_fv_phys.cpp

```diff
@@ -0,0 +1,320 @@
+  const auto& gn = m_phys_grid->name();
+  const auto nlev = get_field_out("T_mid", gn).get_view<Real**>().extent_int(1);
+  const auto nq = get_group_out("tracers").m_bundle->get_view<Real***>().extent_int(1);
+  assert(get_field_out("T_mid", gn).get_view<Real**>().extent_int(0) == nelem*npg);
+  assert(get_field_out("horiz_winds", gn).get_view<Real***>().extent_int(1) == 2);
+  
+  const auto ps = Homme::GllFvRemap::Phys1T(
+    get_field_out("ps", gn).get_view<Real*>().data(),
+    nelem, npg);
+  const auto phis = Homme::GllFvRemap::Phys1T(
+    get_field_out("phis", gn).get_view<Real*>().data(),
+    nelem, npg);
+  const auto T = Homme::GllFvRemap::Phys2T(
+    t ? t->T_mid.data() : get_field_out("T_mid", gn).get_view<Real**>().data(),
+    nelem, npg, nlev);
+  const auto omega = Homme::GllFvRemap::Phys2T(
+    get_field_out("omega", gn).get_view<Real**>().data(),
+    nelem, npg, nlev);
+  const auto uv = Homme::GllFvRemap::Phys3T(
+    t ? t->horiz_winds.data() : get_field_out("horiz_winds", gn).get_view<Real***>().data(),
+    nelem, npg, 2, nlev);
+  const auto q = Homme::GllFvRemap::Phys3T(
+    t ? t->tracers.data() : get_group_out("tracers", gn).m_bundle->get_view<Real***>().data(),
+    nelem, npg, nq, nlev);
+  const auto dp = Homme::GllFvRemap::Phys2T(
+    get_field_out("pseudo_density", gn).get_view<Real**>().data(),
+    nelem, npg, nlev);
+
+  gfr.run_dyn_to_fv_phys(time_idx, ps, phis, T, omega, uv, q, &dp);
+  Kokkos::fence();
+}
+
+void HommeDynamics::remap_fv_phys_to_dyn () const {
+  if (not fv_phys_active()) return;
+  const auto& c = Homme::Context::singleton();
+  auto& gfr = c.get<Homme::GllFvRemap>();
+  const auto time_idx = c.get<Homme::TimeLevel>().n0;
+  constexpr int NGP = HOMMEXX_NP;
+  const int nelem = m_dyn_grid->get_num_local_dofs()/(NGP*NGP);
+  const auto npg = m_phys_grid_pgN*m_phys_grid_pgN;
+  const auto& gn = m_phys_grid->name();
+  const auto nlev = m_helper_fields.at("FT_phys").get_view<const Real**>().extent_int(1);
+  const auto nq = get_group_in("tracers", gn).m_bundle->get_view<const Real***>().extent_int(1);
+  assert(m_helper_fields.at("FT_phys").get_view<const Real**>().extent_int(0) == nelem*npg);
+
+  const auto uv_ndim = m_helper_fields.at("FM_phys").get_view<const Real***>().extent_int(1);
+  assert(uv_ndim == 2 or uv_ndim == 3);
+
+  const auto T = Homme::GllFvRemap::CPhys2T(
+    m_helper_fields.at("FT_phys").get_view<const Real**>().data(),
+    nelem, npg, nlev);
+  const auto uv = Homme::GllFvRemap::CPhys3T(
+    m_helper_fields.at("FM_phys").get_view<const Real***>().data(),
+    nelem, npg, uv_ndim, nlev);
+  const auto q = Homme::GllFvRemap::CPhys3T(
+    get_group_in("tracers", gn).m_bundle->get_view<const Real***>().data(),
+    nelem, npg, nq, nlev);
+  
+  gfr.run_fv_phys_to_dyn(time_idx, T, uv, q);
+  Kokkos::fence();
+  gfr.run_fv_phys_to_dyn_dss();
+  Kokkos::fence();
+}
+
+// [rrtmgp active gases] This is to address issue #1782. It supports option 1 in
```

Comment:
Would it be helpful to add the string "TODO" to this comment so that it is easily identifiable?

Reply:
I can do that. Is "TODO [rrtmgp active gases]" fine? The bracket thing I do just for myself to group related comments in a grep-able way.

Reply:
Absolutely. I like the TODO string because it jumps out at the developer.